test_scripts/Defects/6_1/3142_02_Absence_of_misplaced_OnAudioDataStreaming.lua

```diff
@@ -0,0 +1,89 @@
+---------------------------------------------------------------------------------------------------
+-- Issue: https://github.com/smartdevicelink/sdl_core/issues/3142
+--
+-- Steps:
+-- 1. Set StopStreamingTimeout = 3000 and AudioDataStoppedTimeout = 3000 in SDL .INI file
+-- 2. Start SDL, HMI, connect Mobile device
+-- 3. Register 2 NAVIGATION applications: App_1 and App_2
+-- 4. Activate App_1 and start Audio streaming
+-- 5. Deactivate App_1 (streaming still continue since app has STREAMABLE state)
+-- 6. Activate App_2 and start Audio streaming
+-- 7. App_1 and App_2 continue streaming data within 'StopStreamingTimeout' timeout
+-- SDL does:
+--   - switch streaming between apps and provides HMI with streaming data from App_2
+--   - not unregister App_1 since timeout is not yet expired
+-- 8. App_1 stops streaming within 'StopStreamingTimeout' timeout
+-- SDL does:
+--   - not send Navi.OnAudioDataStreaming(false) notification to HMI since App_2 still continue streaming
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require("test_scripts/Defects/6_1/common_3139_3140_3142")
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Apps Configuration ]]
+common.app.getParams(1).appHMIType = { "NAVIGATION" }
+common.app.getParams(2).appHMIType = { "NAVIGATION" }
```

Comment:
@ShobhitAd Originally `3139` and `3142` issues was found during testing of `PROJECTION` and `NAVIGATION` apps. The main difference is:
1) in case of different HMI types (`PROJECTION` as for app1 and `NAVIGATION` as for app2) if 2nd app is activated, 1st app will be switched to `LIMITED` level
2) in case of the same HMI types (app1 and app2 both have `NAVIGATION`) if 2nd app is activated, 1st app will be switched to `BACKGROUND` level

However `3142` it is reproduced in both cases. So I've updated the description of the issue to show 2nd case (in order to not have a dependency on fix for `3139`)

Reply:
@ShobhitAd Please see comment above